activerecord/test/cases/adapters/mysql2/mysql2_adapter_test.rb

```diff
@@ -71,6 +71,31 @@ def close
     end
   end
 
+  def test_mysql2_default_prepared_statements
+    fake_connection = Class.new do
+      def query_options
+        {}
+      end
+
+      def query(*)
+      end
+
+      def close
+      end
+    end.new
+
+    adapter = assert_deprecated do
```

Comment:
I remembered the other method I was thinking of here:

```suggestion
    adapter = ActiveSupport::Deprecation.silence do
```

This won't fail if the deprecation is missing - that's the job of the test above this one.